### DISCUSSION TITLE
Enable DVCC for what?

### DISCUSSION THREAD

**Discussion by centro-max** [Louisvdw/dbus-serialbattery#139] [Q&A] [answered]
The installation description states that DVCC must be activated. I didn't do this and the driver works fine. My question is, why should DVCC be activated, what changes then and what is the advantage?

**Comment by Louisvdw** [comment, marked as answer]
DVCC - Distributed Voltage and Current Control

Enabling DVCC changes a GX device from a passive monitor into an active controller.
With an Managed BMS battery using the serial connection and driver, the GX receives a Charge Voltage Limit (CVL), Charge Current Limit (CCL), Discharge Current Limit (DCL) from that battery and relays that to the connected inverter/chargers and solar chargers. These then disable their internal charge algorithms and simply do what they're told by the battery. There is no need to set-up charge voltages or choose the charge algorithm type.

In short if you don't enable DVCC then each device will produce power from the settings you have set them up with, while when you enable DVCC it tells the GX to use the settings that the battery provide to manage the system. 

Example: You have set you MPPT to charge the battery with up to 70A and to a voltage of 56V. If you connect this to a battery that can only handle 50A and a voltage of 55V it will not know about this and the MPPT will charge to the values it was set up with.
If you enable DVCC the battery can tell the system it will only accept up to 50A charge current and the MPPT will override it's internal value and honor the 50A. Same with the the voltage which it will aim for 55V.

What the driver does is to use the values from the battery to keep the system inside the parameters that the battery can handle so that the battery does not need to disconnect itself to protect the cells (which means your power cuts out)

With DVCC disabled you will still see the values that the battery publish to the GX, but it will not be used for anything. It's just a display of numbers.

https://www.victronenergy.com/media/pg/CCGX/en/dvcc---distributed-voltage-and-current-control.html

**Reply by centro-max** [reply]
Many thanks for the explanation. However, I have found that even if DVCC is deactivated, at least the value for the Charge Current Limit (CCL) is accepted. That's definitely the case, my 3 * MultiPlus II 3000/48 only charge up to the current CCL value in the serial battery driver - and that with deactivated DVCC. I tested and checked it with different values. That`s why i opened this question.

**Reply by Louisvdw** [reply]
Intresting. I'll dig a bit deeper when I get some time

**Comment by Panamabu** [comment]
I have another issue with the DVCC that i hope can address here correctly. I have one MP2 5000/48 with a 2P16S 280ah EVE cell pack, each 16S pack with its own Daly BMS and additional each pack with one NEEY 4A active balancer. i would like to keep the DVCC enabled, but the values, especially for discharging the battery makes it terribly slow. since i have 2 cell packs i should at least be able to double the allowed Amps for charging and discharging at their limits. Where does the Values for the DVCC come from? is there any possibility to adjust them to my own preference to modify the charging and discharging curves how i like it... at the current state from 10%SOC below, i can only take like 150W out of the battery, even there is still more than 2-3 kwh left of useable capacity...

**Reply by Louisvdw** [reply]
https://github.com/Louisvdw/dbus-serialbattery/wiki/How-to-install#how-to-increase-the-default-limits

You can also disable CCCM by setting CCCM_ENABLE = False

**Reply by Panamabu** [reply]
yes i did and adjusted it to the 100A // 200A setting of the BMS, however the allowed amps for discarging below 10% is rediculously low. can i change the setpoimts somewhere?

**Reply by eliasgraf** [reply]
The limits can be adjusted in the battery.py script, after that you have to restart the driver/reboot.

**Reply by pau1phi11ips** [reply]
> The limits can be adjusted in the battery.py script

I think it' the `utils.py` file that holds the values.
